core/components/menu/menu.js

```diff
@@ -77,27 +83,57 @@ Blockly.Menu = function() {
    * @type {?Blockly.EventData}
    * @private
    */
-  this.onKeyDownWrapper_ = null;
+  this.onKeyDownHandler_ = null;
 };
-Blockly.utils.object.inherits(Blockly.Menu, Blockly.Component);
 
 
+/**
+ * Add a new menu item to the bottom of this menu.
+ * @param {!Blockly.MenuItem} menuItem Menu item to append.
+ */
+Blockly.Menu.prototype.addChild = function(menuItem) {
+  this.menuItems_.push(menuItem);
+};
+
 /**
  * Creates the menu DOM.
- * @override
+ * @param {!Element} container Element upon which to append this menu.
  */
-Blockly.Menu.prototype.createDom = function() {
+Blockly.Menu.prototype.render = function(container) {
   var element = document.createElement('div');
-  element.id = this.getId();
-  this.setElementInternal(element);
-
-  // Set class
-  element.className = 'goog-menu goog-menu-vertical blocklyNonSelectable';
+  // goog-menu is deprecated, use blocklyMenu.  May 2020.
+  element.className = 'blocklyMenu goog-menu blocklyNonSelectable';
   element.tabIndex = 0;
+  Blockly.utils.aria.setRole(element, this.roleName_);
```

Comment:
Any reason to not default to a role name of menu here?

Reply:
We currently use this code for two things, menus and dropdowns, each of which have completely different roles.  I'd rather the role be set explicitly since otherwise developers copying the menu code won't see the setRole method and may end up with UI that quietly has the wrong role.

Reply:
Fair enough.